src/document.rs

```diff
@@ -1,9 +1,11 @@
-use tendril::StrTendril;
+use tendril::{StrTendril, ByteTendril, ReadExt};
 
 use node::{self, Node};
 use predicate::Predicate;
 use selection::Selection;
 
+use std::io::{Read, Error, Result, ErrorKind};
```

Comment:
One more thing - could you not import `Result` here (maybe just import `std::io` and use `io::...` everywhere; that's what I've seen most people do)? `Result` is already in the prelude and this overrides it.